File: split.js
const fs = require('fs');
const path = require('path');

const file = path.join(__dirname, 'index.html');
let content = fs.readFileSync(file, 'utf8');

// Extract CSS
const styleRegex = /<style>([\s\S]*?)<\/style>/;
const styleMatch = content.match(styleRegex);
if (styleMatch) {
  const cssContent = styleMatch[1];
  
  // Split CSS
  // Rough split for now, we will save it all to base.css and split later if needed,
  // or we can just write it to css/main.css
  fs.writeFileSync(path.join(__dirname, 'css/main.css'), cssContent.trim());
  content = content.replace(styleRegex, '<link rel="stylesheet" href="./css/main.css" />');
}

// Extract JS
// There are multiple script tags. We can extract them sequentially.
const scripts = [];
let scriptMatch;
const scriptRegex = /<script.*?>([\s\S]*?)<\/script>/g;

let jsContent = '';
while ((scriptMatch = scriptRegex.exec(content)) !== null) {
  if (scriptMatch[1].trim() !== '') {
     jsContent += scriptMatch[1].trim() + '\n\n';
  }
}

// Just write all JS to js/main.js for now, then we'll clean it up
if (jsContent) {
   fs.writeFileSync(path.join(__dirname, 'js/main.js'), jsContent);
}

// We will manually replace the script tags later because of importmaps, etc.
// But we can remove the old <script> content
let newContent = content.replace(/<script>([\s\S]*?)<\/script>/g, '');
newContent = newContent.replace(/<script type="module">([\s\S]*?)<\/script>/g, '<script type="module" src="./js/main.js"></script>');

fs.writeFileSync(path.join(__dirname, 'index_split.html'), newContent);
console.log("Done splitting.");

File: split.py
import re
import os

with open('index.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract styles
style_match = re.search(r'<style>([\s\S]*?)</style>', content)
if style_match:
    with open('css/main.css', 'w', encoding='utf-8') as f:
        f.write(style_match.group(1).strip())
    content = content.replace(style_match.group(0), '<link rel="stylesheet" href="./css/main.css" />')

# Extract scripts
# Find all script tags except importmaps
scripts = re.findall(r'<script.*?>([\s\S]*?)</script>', content)
js_content = ""
for s in scripts:
    if s.strip():
        # don't extract JSON importmap
        if '"imports":' not in s:
            js_content += s.strip() + "\n\n"

with open('js/main.js', 'w', encoding='utf-8') as f:
    f.write(js_content)

# We will remove the script bodies manually or via script
content = re.sub(r'<script>[\s\S]*?</script>', '', content)
content = re.sub(r'<script type="module">[\s\S]*?</script>', '<script type="module" src="./js/main.js"></script>', content)

with open('index.html', 'w', encoding='utf-8') as f:
    f.write(content)

print("Split completed successfully!")
